### PR TITLE
[storage/mmr] change anydb::variable max_ops to NonZeroU64

### PR DESCRIPTION
**Pull request by roberto-bayardo**
Change max_ops in proof generation in adb::variable to NonZeroU64 to match the other adb proving APIs and prevent generation of an unverifiable empty proof, and disallow proofs over empty ranges / element lists more broadly.

Multi_proving over empty element lists was special cased to have an empty list of digests but non-zero size for non-empty databases, whereas empty range proofs would include the digests of every MMR peak (allowing root reconstruction if it were to be handled properly).  Neither approach is particuarly appealing, and it feels better to simply require you always be proving inclusion of *something*.

Tweaked and added a few more tests around empty proof verification. Empty proofs should only verify against a completely empty mmr and nothing else.

### CODE REVIEW

File: storage/src/mmr/proof.rs [hunk 1]
```diff
@@ -534,6 +538,7 @@ pub(crate) fn nodes_required_for_multi_proof(
     // Collect all required node positions
     //
     // TODO(#1472): Optimize this loop
+    assert!(!locations.is_empty(), "empty locations");
```
Comment (patrick-ogrady) [inline storage/src/mmr/proof.rs:541]: I'm assuming there is a "precheck" call that prevents us from hitting this unless called directly (following the error changes).

Comment (patrick-ogrady) [reply storage/src/mmr/proof.rs:541]: Ah yep, found it.
